### DISCUSSION TITLE
Keeping the dependency set small.

### DISCUSSION THREAD

**Discussion by vchuravy** [FedeClaudi/Term.jl#48] [General]
I was thinking about if I can use Term.jl for Cthulhu.jl (since I need things like pagination) and I was auditing the dependency chain. Is Documenter.jl really needed? It seems like a large dependency.

**Comment by FedeClaudi** [comment]
Hi,  that's great!

`Documenter` is used to perform [code introspection](https://fedeclaudi.github.io/Term.jl/dev/adv/introspection/) [here](https://github.com/FedeClaudi/Term.jl/blob/7b87badf49eae15cd172e0b72be88fee8c2d96d8/src/_inspect.jl#L1)

Essentially I need something that can easily get me the docstring of a Type or method. If there's an alternative to that I'd be happy to remove documenter from the requirements. 
Alternatively perhaps it can be made an optional requirement. but I don't know how to do that in Julia?

**Comment by vchuravy** [comment]
The docsystem in base should implement that functionality?

```
julia> @doc sin
```

In general:

```
help?> Base.Docs.doc
  Docs.doc(binding, sig)

  Return all documentation that matches both binding and sig.

  If getdoc returns a non-nothing result on the value of the binding, then a dynamic docstring is returned instead
  of one based on the binding itself.
```

**Comment by FedeClaudi** [comment]
Thank's, I'll give it a try and if it works the next version won't need Documenter.

**Comment by FedeClaudi** [comment]
Hi, 
https://github.com/FedeClaudi/Term.jl/pull/60 introduces changes to Term's introspection code to avoid needing `Documenter` as a dependency.  Thanks for the advice!
